pkg/controllerhelpers/scylla.go

```diff
@@ -15,46 +16,110 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/errors"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	corev1schedulinghelpers "k8s.io/component-helpers/scheduling/corev1"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
 )
 
-func GetScyllaIPFromService(svc *corev1.Service) (string, error) {
-	if svc.Spec.Type != corev1.ServiceTypeClusterIP {
-		return "", fmt.Errorf("service %s is of type %q instead of %q", naming.ObjRef(svc), svc.Spec.Type, corev1.ServiceTypeClusterIP)
+func GetScyllaHost(ctx context.Context, sc *scyllav1.ScyllaCluster, coreV1Client corev1client.CoreV1Interface, svc *corev1.Service) (string, error) {
```

Comment:
Maybe we should make a function that gets a tuple client+node IPs and use the same in one in member configuration

Reply:
>  a function that gets a tuple client+node IPs

how would that work? I can see it taking service and pod objects but don't see how this could work with the addresses directly

Reply:
i wrote the "gets" from the callers point of view, so the function would return the tuple.